pennylane/circuit_drawer.py

```diff
@@ -0,0 +1,916 @@
+
+        return self
+
+    def append_wire(self, wire):
+        """Append a wire to the Grid.
+
+        Args:
+            wire (list): Wire that will be appended
+        """
+        self.raw_grid.append(wire)
+        self.raw_grid_transpose = _transpose(self.raw_grid)
+
+        return self
+
+    @property
+    def num_layers(self):
+        """Number of layers in the Grid.
+
+        Returns:
+            int: Number of layers in the Grid
+        """
+        return len(self.raw_grid_transpose)
+
+    def layer(self, idx):
+        """Return the layer at the specified index.
+
+        Args:
+            idx (int): Index of the layer to be retrieved
+
+        Returns:
+            list: The layer at the specified index
+        """
+        return self.raw_grid_transpose[idx]
+
+    @property
+    def num_wires(self):
+        """Number of wires in the Grid.
+
+        Returns:
+            int: Number of wires in the Grid
+        """
+        return len(self.raw_grid)
+
+    def wire(self, idx):
+        """Return the wire at the specified index.
+
+        Args:
+            idx (int): Index of the wire to be retrieved
+
+        Returns:
+            list: The wire at the specified index
+        """
+        return self.raw_grid[idx]
+
+    def copy(self):
+        """Create a copy of the Grid.
+
+        Returns:
+            Grid: A copy of the Grid
+        """
+        return Grid(self.raw_grid.copy())
+
+    def append_grid_by_layers(self, other_grid):
+        """Append the layers of another Grid to this Grid.
+
+        Args:
+            other_grid (pennylane.circuit_drawer.Grid): Grid whos layers will be appended
+        """
+        for i in range(other_grid.num_layers):
+            self.append_layer(other_grid.layer(i))
+
+        return self
+
+    def __str__(self):
+        """String representation"""
+        ret = ""
+        for wire in self.raw_grid:
+            ret += str(wire)
+            ret += "\n"
+
+        return ret
+
+
+class CharSet(abc.ABC):
+    """Charset base class."""
+
+    # pylint: disable=too-few-public-methods
+
+    WIRE = None
+    MEASUREMENT = None
+    TOP_MULTI_LINE_GATE_CONNECTOR = None
+    MIDDLE_MULTI_LINE_GATE_CONNECTOR = None
+    BOTTOM_MULTI_LINE_GATE_CONNECTOR = None
+    EMPTY_MULTI_LINE_GATE_CONNECTOR = None
+    CONTROL = None
+    LANGLE = None
+    RANGLE = None
+    VERTICAL_LINE = None
+    CROSSED_LINES = None
+    PIPE = None
+    OTIMES = None
+
+    @staticmethod
+    def to_superscript(num):
+        """Convert the given number to a superscripted string."""
+        pass
+
+
+class UnicodeCharSet(CharSet):
+    """Charset for CircuitDrawing made of Unicode characters."""
+
+    # pylint: disable=too-few-public-methods
+
+    WIRE = "─"
+    MEASUREMENT = "┤"
+    TOP_MULTI_LINE_GATE_CONNECTOR = "╭"
+    MIDDLE_MULTI_LINE_GATE_CONNECTOR = "├"
+    BOTTOM_MULTI_LINE_GATE_CONNECTOR = "╰"
+    EMPTY_MULTI_LINE_GATE_CONNECTOR = "│"
+    CONTROL = "C"
```

Comment:
Just out of curiosity, did you consider this one for control qubits?
https://www.fileformat.info/info/unicode/char/25cf/browsertest.htm

Reply:
Yes, it didn't look good